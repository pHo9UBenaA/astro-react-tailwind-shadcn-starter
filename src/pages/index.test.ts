import preactRenderer from '@astrojs/preact/server.js';
import { experimental_AstroContainer as AstroContainer } from 'astro/container';
import { expect, test } from 'vitest';
import Index from './index.astro';

expect.addSnapshotSerializer({
	test: (val) => typeof val === 'string',
	print: (val) => {
		return (
			(val as string)
				// Remove `data-astro-source-file` attributes to ensure consistent snapshots across different environments.
				// see: https://github.com/pHo9UBenaA/astro-react-tailwind-shadcn-starter/actions/runs/13352440761/job/37290691107?pr=10
				.replace(/\s+data-astro-source-file="[^"]*"/g, '')
		);
	},
});

test('snapshot', async () => {
	const container = await AstroContainer.create();

	// see: https://github.com/withastro/astro/issues/11882#issuecomment-2320538563
	container.addServerRenderer({
		name: '@astro/preact',
		renderer: preactRenderer,
	});

	container.addClientRenderer({
		name: '@astrojs/preact',
		entrypoint: '@astrojs/preact/client.js',
	});

	const result = await container.renderToString(Index);

	expect(result).toMatchSnapshot();
});
