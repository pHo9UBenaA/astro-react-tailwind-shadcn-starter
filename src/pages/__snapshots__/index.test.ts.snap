// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `<html lang="en"> <head><meta charset="UTF-8"><meta name="robots" content="index, follow"><meta name="viewport" content="width=device-width"><title>Astro Basics</title><meta name="description" content="Astro Basics"><link rel="icon" type="image/svg+xml" href="/favicon.svg"><meta property="og:title" content="Astro Basics"><meta property="og:description" content="Astro Basics"><meta property="og:type" content="website"><meta property="og:url" content="https://example.com/"><meta property="og:image" content="https://example.com//ogp.png"><meta name="author" content="Komiya"></head> <body data-astro-source-loc="26:8">  </body></html> <div class="flex justify-center items-center h-screen" data-astro-source-loc="7:60">  </div> `;
